### PR TITLE
Cover asset_caching by tests

### PR DESCRIPTION
**Pull request by okurz**
Extracting method do_asset_caching for easier testing.

### CODE REVIEW

**Comment by okurz** [comment]
Trying again ;)

File: t/24-worker-engine.t [hunk 1]
```diff
@@ -93,4 +95,36 @@ subtest 'asset settings' => sub {
...
+    my $mojo_file = Test::MockModule->new('Mojo::File');
+    $mojo_file->mock(path => undef);
+    my $cache_request = Test::MockModule->new('OpenQA::Worker::Cache::Request');
+    $cache_request->mock(enqueue => undef);
+    $cache_request->mock(asset => sub { return OpenQA::Worker::Cache::Request->new });
+    $got = OpenQA::Worker::Engines::isotovideo::cache_assets(undef, $settings, \%assets);
+    like($got->{error}, qr/Failed to download/, 'cache_assets can not pick up supplied assets when not found') or diag explain $got;
+};
+
+$got = OpenQA::Worker::Engines::isotovideo::do_asset_caching($settings);
```
Comment (Martchus) [inline t/24-worker-engine.t]: Can you make this a subtest, too? And I miss the place where `$got` is declared/defined. I only see `my $got` in nested scopes.

File: t/24-worker-engine.t [hunk 2]
```diff
@@ -93,4 +95,36 @@ subtest 'asset settings' => sub {
...
 
 
+subtest 'caching' => sub {
+    is(OpenQA::Worker::Engines::isotovideo::cache_assets, undef, 'cache_assets has nothing to do without assets');
+    my %assets = (
+        ISO => 'foo.iso',
+    );
+    my $cache_client = Test::MockModule->new('OpenQA::Worker::Cache::Client');
+    $cache_client->mock(availability_error => undef);
+    #$cache_client->mock(asset_path => undef);
```
Comment (Martchus) [inline t/24-worker-engine.t]: An out-commented line (with no explanation).

**Comment by kraih** [comment]
Yes, this conflicts with what i'm working on currently.

**Comment by kraih** [comment]
This PR conflicts with #2434.

**Comment by kraih** [comment]
And with #2441. There might still be smaller API changes following to improve cache service error logs.

**Comment by okurz** [comment]
I will rebase the tests and bring it up again, possibly after more changes by kraih

**Comment by kraih** [comment]
Yes, i'm still making changes to the code you want to test. Might be best to just wait for now.

**Pull request closed without merging**